File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        // for (int i=0; i<nums.size(); i++){
        //     for (int j=0; j<i; j++){
        //         if(nums[j]==nums[i]) return true;
        //     }
        // }
        // return false;

        sort(nums.begin(), nums.end());
        for(int i=1; i<nums.size(); i++){
            if(nums[i-1]==nums[i]) return true;
        }
        return false;
    }
};

File: Data Structures & Algorithms/continuous-subarray-sum/submission-1.py
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        mp = {0: -1}  # stores rem -> idx
    
        prefix = 0

        for i, n in enumerate(nums):
            prefix += n
            rem = prefix % k

            if rem in mp:
                if i - mp[rem] >= 2:
                    return True
            else:
                mp[rem] = i

        return False